include/glow/Base/Type.h

```diff
@@ -132,14 +132,18 @@ struct PaddingTLBR {
   }
 };
 
-/// Collapse a tensor shape into two sizes: the first dimension and the size of
-/// the rest of the dimensions.
-/// For example, [7, 3, 4, 2] -> [7, 24]
-inline std::pair<size_t, size_t> flattenCdr(llvm::ArrayRef<size_t> dims) {
-  assert(dims.size() > 1);
+/// Collapse a tensor shape into two sizes: the first n dimensions and the size
+/// of the rest of the dimensions. For example, ([7, 3, 4, 2], 1) -> [7, 24]
+inline std::pair<size_t, size_t> flattenCdr(llvm::ArrayRef<size_t> dims,
+                                            size_t n = 1) {
+  assert((dims.size() > 1 && n < dims.size()));
```

Comment:
No need in extra brackets.

Also, please check that `1 <= n`. Then `dims.size() > 1` check is not needed, because `1 <= n && n < dims.size()` implies that `dims.size()` is at least 2.